### DISCUSSION TITLE
Final type for dev::Transform

### DISCUSSION THREAD

**Discussion by johnlepikhin** [actix/actix-web#3174] [Q&A]
Hi there,

I'm trying to implement an external generator of middlewares. Somewhere in modules, `Vec<Transform>` is generated and then passed into a function which will build the application. But I gave up to construct some final type of `Transform<>` with all correct type arguments. Can you give some example? Solution based on `.wrap_fn()` or experimental `.from_fn()` also would be helpful.

Thank you!

**Comment by johnlepikhin** [comment]
It seems every `Transform<>` has its own unique type parameters, and each `.wrap()` returns some new type of `App<>`. Thus, probably it's impossible to implement such architecture. Am I right?

**Reply by robjtede** [reply]
It should be somewhat possible to build this sort of architecture if you box everything to a homogenious type in the Vec. Maybe something similar to the [`Compat`](https://docs.rs/actix-web/latest/actix_web/middleware/struct.Compat.html) middleware wrapper.

I could be wrong and it's not possible right now though. Can you sort of write down what code you _want_ to be able to write.
